stdlib/public/TensorFlow/TensorHandle.swift

```diff
@@ -97,15 +97,25 @@ extension TensorHandle : TensorSendableReceivable {
   ) -> TensorHandle<Scalar> {
     debugLog("Receiving tensor of id \(tensorId) and type \(Scalar.self).")
     let status = TF_NewStatus()
-    let cTensor: CTensor? = TF_DequeueNamedTensor(
-      computation.cSession, Int32(tensorId), status)
-    checkOk(status)
-    internalConsistencyCheck(
-      cTensor != nil,
-      "TF_DequeueNamedTensor() cannot return nil when the status is OK.")
-    TF_DeleteStatus(status)
-    let tensorHandle = TensorHandle<Scalar>(copyingFromCTensor: cTensor!)
-    TF_DeleteTensor(cTensor!)
+    internalConsistencyCheck(status != nil)
+    let tensorHandle: TensorHandle<Scalar>
+    if _RuntimeConfig.usesTFEagerAPI {
+      let context = _ExecutionContext.global
+      let cTensorHandle = TFE_DequeueNamedTensorFromCtx(
+        context.eagerCtx, Int32(tensorId), Scalar.cDataType, status)
+      checkOk(status)
+      tensorHandle = TensorHandle<Scalar>(owning: cTensorHandle!)
+    } else {
+      let cTensor: CTensor? = TF_DequeueNamedTensor(
```

Comment:
This forms a perfect use case for implicitly unwrapped optionals. You can do
```swift
let cTensor: CTensor! = ...
checkOk(status)
internalConsistencyCheck(cTensor != nil, ...)
tensorHandle = TensorHandle<Scalar>(copyingFromCTensor: cTensor) // no ! needed here and below
TF_DeleteTensor(cTensor)
```

Since this is just some old code being moved around, it's fine too keep it as is!

Reply:
hmm, if TF_DequeueNamedTensor() returns NULL, wouldn't `let cTensor: CTensor! = ...` crash?

Reply:
It would not crash, because IUOs are only unwrapped when you pass them around as a non-optional. `let cTensor: CTensor!` is just an initialization, not passing.

In other words, IUOs are just optionals that get implicitly unwrapped when you use them in positions where the contextual type is not optional.

Reply:
Thanks for clarifying. I changed the code to use IUOs. The code is indeed simpler!